### PR TITLE
i3status: add module

### PR DESCRIPTION
**Pull request by offlinehacker**

### CODE REVIEW

**Comment by offlinehacker** [comment]
Any updates on getting this merged?

File: modules/programs/i3status.nix [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+          opts = mkOption {
+            description = "Options for i3status block";
+            type = types.attrs;
+            default = {};
+          };
+        };
+      }));
+
+      example = literalExample ''{
```
Comment (rycee) [inline modules/programs/i3status.nix]: I think this example is too long. Would it be possible to reduce it to a few lines?

Also, placing the `{` directly after `''` will make the indentation look strange in the documentation.

File: modules/programs/i3status.nix [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  '';
+in {
+  options.programs.i3status = {
+    enable = mkEnableOption "i3status: Generates status bar to use with i3bar, dzen2 or xmobar.";
+
+    order = mkOption {
+      description = "Order of i3status blocks.";
+      type = types.listOf types.str;
+      default = [];
+      example = literalExample ''[
```
Comment (rycee) [inline modules/programs/i3status.nix]: The indentation is wrong in the documentation with `[` directly after `''`: 
i3status

Can do something like
```
example = literalExample ''
  [
    (mkOrder 500 "online_status")
    (mkOrder 510 "disk")
    (mkOrder 520 "load")
    (mkOrder 530 "net_rate")
    (mkOrder 540 "volume master")
  ]
'';
```
instead.

File: modules/programs/i3status.nix [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        (mkOrder 510 "disk")
+        (mkOrder 520 "load")
+        (mkOrder 530 "net_rate")
+        (mkOrder 540 "volume master")
+      ]'';
+    };
+
+    blocks = mkOption {
+      description = "Attribute set of defined i3status blocks.";
+      type = types.attrsOf (types.submodule ({name, config, ...}: {
```
Comment (rycee) [inline modules/programs/i3status.nix]: The `config` argument is not used, as far as I can tell. Perhaps safe to remove?

File: modules/programs/i3status.nix [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+let
+  cfg = config.programs.i3status;
+
+  toStr = val:
+    if isString val then ''"${val}"''
+    else if isBool val then (if val then "true" else "false")
+    else toString val;
+
+  blockToConfig = block: ''${block.type} ${optionalString (block.name != null) ''"${block.name}"''} {
```
Comment (rycee) [inline modules/programs/i3status.nix]: Can split the line to make it a bit shorter:
```suggestion
  blockToConfig = block: ''
    ${block.type} ${optionalString (block.name != null) ''"${block.name}"''} {
```

**Comment by vojta001** [comment]
Hi @offlinehacker. `i3status` configuration via `home-manager` is something I would really like to see implemented. Are you still willing to finish it (implement @rycee's suggestions)? If not, I am offering I will do it.

**Comment by offlinehacker** [comment]
Sorry it took so long, suggested changes are now implemented.

**Pull request closed without merging**